Source/VCGeneration/Split.cs

```diff
@@ -868,6 +1036,65 @@ public Counterexample ToCounterexample(ProverContext context)
         return newBlocks;
       }
 
+      private static List<Block> PostProcess(List<Block> blocks)
+      {
+        void DeleteFalseGotos (Block b)
+        {
+          bool isAssumeFalse (Cmd c) { return c is AssumeCmd ac && ac.Expr is LiteralExpr le && !le.asBool; }
+          var firstFalseIdx = b.Cmds.FindIndex(c => isAssumeFalse(c));
+          if (firstFalseIdx != -1)
+          {
+            b.Cmds = b.Cmds.Take(firstFalseIdx + 1).ToList();
+            b.TransferCmd = (b.TransferCmd is GotoCmd) ? new ReturnCmd(b.tok) : b.TransferCmd;
+          }
+        }
+
+        void DeleteSubSumptions (Block b)
+        {
+          bool isAssumeSubsumption (Cmd c) { return c is AssumeCmd ac && QKeyValue.FindBoolAttribute(ac.Attributes, "subsumption"); }
+          b.Cmds = b.Cmds.Where(c => !isAssumeSubsumption(c)).ToList();
```

Comment:
`{:subsumption 0}` is something that can be placed on an `AssertCmd`. This code looks for `:subsumption` placed on an `AssumeCmd`. I'm guessing this comes about because an `AssertCmd` has been turned into an `AssumeCmd`. Is it possible to instead never create such an `AssumeCmd` from an `assert {:subsumption 0} ...`?

Reply:
Also, I think it's best to also check for the the `0` argument of `:subsumption`, even though I don't think anyone ever uses `1` or `2`.

Reply:
`VCGen.AssertTurnedIntoAssume` already does it! No need for this piece of code.